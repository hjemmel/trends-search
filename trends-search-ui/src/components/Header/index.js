import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import {
  Alignment,
  Classes,
  Button,
  Navbar,
  NavbarGroup,
  NavbarHeading
} from '@blueprintjs/core';

class Header extends React.Component {
  render (props) {
    const classes = classNames(Classes.DARK, Classes.FIXED_TOP);
    return (
      <div className={this.props.className}>
        <Navbar className={classes}>
          <NavbarGroup align={Alignment.LEFT}>
            <NavbarHeading>Trends Analyzer</NavbarHeading>
          </NavbarGroup>
          <NavbarGroup align={Alignment.RIGHT}>
            <Button className={Classes.MINIMAL} icon='user' />
            <Button className={Classes.MINIMAL} icon='notifications' />
            <Button className={Classes.MINIMAL} icon='cog' />
          </NavbarGroup>
        </Navbar>
      </div>
    );
  }
}

Header.propTypes = {
  className: PropTypes.string
};

export default Header;
