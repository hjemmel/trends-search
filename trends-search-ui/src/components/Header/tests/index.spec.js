import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../index';

describe('<Header />', () => {
  it('should render properly', () => {
    const renderedComponent = shallow(<Header />);
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });

  it('should render with className', () => {
    const renderedComponent = shallow(<Header className='morty' />);
    expect(renderedComponent.props().className).toBe('morty');
    renderedComponent.setProps({ className: 'rick' });
    expect(renderedComponent.props().className).toBe('rick');
  });

  it('should render the header', () => {
    const renderedComponent = shallow(<Header />);
    expect(renderedComponent.contains('Trends Analyzer')).toBeTruthy();
  });
});
