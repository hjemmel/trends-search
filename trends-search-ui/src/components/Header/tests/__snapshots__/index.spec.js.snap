// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render properly 1`] = `
<div>
  <Blueprint2.Navbar
    className="bp3-dark bp3-fixed-top"
  >
    <Blueprint2.NavbarGroup
      align="left"
    >
      <Blueprint2.NavbarHeading>
        Trends Analyzer
      </Blueprint2.NavbarHeading>
    </Blueprint2.NavbarGroup>
    <Blueprint2.NavbarGroup
      align="right"
    >
      <Blueprint2.Button
        className="bp3-minimal"
        icon="user"
      />
      <Blueprint2.Button
        className="bp3-minimal"
        icon="notifications"
      />
      <Blueprint2.Button
        className="bp3-minimal"
        icon="cog"
      />
    </Blueprint2.NavbarGroup>
  </Blueprint2.Navbar>
</div>
`;
